frame/evm/src/mock.rs

```diff
@@ -118,6 +118,14 @@ impl FindAuthor<H160> for FindAuthorTruncated {
 	}
 }
 
+/// Returns the Substrate block hash by number.
+pub struct SubstrateBlockHashMapping<T>(sp_std::marker::PhantomData<T>);
```

Comment:
This was not the idea. `SubstrateBlockHashMapping` is included in `pallet-evm` for projects that do not integrate `pallet-ethereum` and need to fall back to the environmental substrate block hash in the runner.

Reply:
Thanks for your explanation. I agree with you, just keep `SubstrateBlockHashMapping` in `pallet-evm`.